### DISCUSSION TITLE
Release 0.9.72 discussion

### DISCUSSION THREAD

**Discussion by kmk3** [netblue30/firejail#5192] [General]
Release links:

* Project board: https://github.com/users/netblue30/projects/4
* GitHub Release: https://github.com/netblue30/firejail/releases/tag/0.9.72

Discussion for the previous release:

* #5181

Discussion for the next release:

* https://github.com/netblue30/firejail/discussions/5597

**Comment by rusty-snake** [comment]
### Blockers

- [x] https://github.com/netblue30/firejail/issues/5533#issuecomment-1357948268

**Comment by kmk3** [comment]
So an RC1 was created yesterday:

* <https://github.com/netblue30/firejail/releases/tag/0.9.72rc1>

@netblue30 When is the intended release date?

Depending on how much time is available, I might try sending a few more fixes.

**Comment by netblue30** [comment]
I was thinking about the second week of January. We might be able to push it towards the end of the month. Debian goes into soft freeze on Feb 12, we should be done ahead of that. So, dump in everything you have: code, new features, bugs, profiles etc. - business as usual!

After the release we start merging in landlock from @ChrysoliteAzalea - either on a branch or directly on the mainline.

**Comment by ghost** [comment]
Interesting times. Lots of security developments happening in Linux land. Very much looking forward to the `landlock` feature.

Some things I've put aside for reading during end-of-year holidays:
- [tetragon](https://github.com/cilium/tetragon): eBPF-based Security Observability and Runtime Enforcement developed by some kernel developers.
- LSM [stackable security modules](https://lwn.net/Articles/912775/).

**Comment by netblue30** [comment]
I'll look into them, thanks!

**Comment by netblue30** [comment]
I'm cleaning up some things here and there, so continue checking in. Final test and archive build on Saturday.

Problems so far:
--whitelist in /dev directory is broken. I'll probably disable it for the next release.

**Reply by kmk3** [reply]
@netblue30 [on Jan 12](https://github.com/netblue30/firejail/discussions/5192#discussioncomment-4668705):

> I'm cleaning up some things here and there, so continue checking in. Final
> test and archive build on Saturday.

For reference, there is the URL opening issue from #5566 (see also the
discussion in #5574), though it's not clear how reproducible it is nor when it
was introduced.

Other than that (and the /dev whitelisting issue below), I can't think of any
blockers and I think that everything is in the RELNOTES, so it looks ready for
releasing to me.

> Problems so far: --whitelist in /dev directory is broken. I'll probably
> disable it for the next release.

Could you clarify what exactly is broken?

It seems to work as intended:

```console
$ firejail --quiet --noprofile --whitelist=/dev/null find /dev
/dev
/dev/null
```

Is the issue related to new devices not showing up in /dev with `private-dev`?
(That is, devices that are connected after the sandbox was started)

Anyway, disabling whitelisting in /dev seems like a rather big change to do at
the last minute.  How about delaying the release for a week to debug this
instead?

**Comment by ghost** [comment]
@netblue30 All I had in my notes that could be relevant for the upcoming release is in our GitHub system. Either merged or in open PR's (to be reviewed). I did notice the option `apparmor-replace` is currently not mentioned in the man page. It is only used in firefox-common.profile:

https://github.com/netblue30/firejail/blob/ee89a263e28902f723a1ac322221f85d9eb2ad24/etc/profile-a-l/firefox-common.profile#L37-L39

https://github.com/netblue30/firejail/blob/ee89a263e28902f723a1ac322221f85d9eb2ad24/src/firejail/sandbox.c#L134-L136

Is this something we should add to the man page nonetheless?

**Comment by netblue30** [comment]
@kmk3

Thanks for the example, I was trying without --noprofile
`````
$ firejail --quiet --whitelist=/dev/null find /dev
fexecve: No such file or directory
Error: failed to run /run/firejail/lib/fseccomp, exiting...
Error: proc 64143 cannot sync with peer: unexpected EOF
Peer 64144 unexpectedly exited with status 1
`````
Probably we pick up something from one of the profiles. The code is still active, the "fix" was to modify the man page. We'll bring a real fix in the release after this one. Anyway, not too many people are using whitelist /dev.

@glitsj16 - we'll have to do some more work on apparmor in the next release, for now undocumented should be fine.

**Reply by rusty-snake** [reply]
```
firejail --whitelist=/dev/null find /dev
```

WFM

Firejail: ee89a26
OS: Fedora 37 / `Linux 6.0.18-300.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Jan 7 17:10:00 UTC 2023`
glibc: 2.36

**Reply by ghost** [reply]
Same here, it's working on Arch Linux with latest git firejail and glibc 2.36.

**Reply by netblue30** [reply]
Debian stable here, kernel 5.10.0-10-amd64, GLIBC 2.31

**Reply by rusty-snake** [reply]
> execveat() was added to Linux in kernel 3.19.  Library support was added to glibc in version 2.34.

**Reply by rusty-snake** [reply]
If `fexecve` started to use `execveat` at the same time we know the reason.

**Reply by reinerh** [reply]
It works for me as well on Debian stable (kernel 5.10.127, glibc 2.31) with firejail 0.9.70:
```
$ firejail --quiet --whitelist=/dev/null find /dev
/dev
/dev/null
```

**Reply by netblue30** [reply]
Thanks, I'll revert the man page.

**Comment by smitsohu** [comment]
In order to work around #5010 and #5230, should we have an opt-in/opt-out option or a toggle in firejail.config?

**Reply by netblue30** [reply]
Maybe toggle in firejail.config?

**Comment by netblue30** [comment]
This is where I'm at:

Debian stable 64 bits - all fine if I disable restrict namespaces, breaks otherwise. Works fine on @reinerh and several other distros:
`````
$  firejail --ignore=restrict-namespaces --whitelist=/dev/null
Reading profile /etc/firejail/default.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-programs.inc

** Note: you can use --noprofile to disable default.profile **

Parent pid 99861, child pid 99862
Child process initialized in 123.10 ms
`````

Debian old-stable 32 bit - simple /bin/bash sandbox, breaks in restict-namespaces, all fine if I --ignore=restrict-namespaces:
`````
$ firejail --debug
[...]
Build restrict-namespaces filter
sbox run: /run/firejail/lib/fseccomp restrict-namespaces.32 /run/firejail/mnt/seccomp/seccomp.namespaces.32 cgroup,ipc,net,mnt,pid,time,user,uts
Dropping all capabilities
Drop privileges: pid 6, uid 1000, gid 1000, force_nogroups 1
No supplementary groups
fseccomp: seccomp_file.c:26: write_to_file: Assertion `size' failed.
Error: failed to run /run/firejail/lib/fseccomp, exiting...
Error: proc 807 cannot sync with peer: unexpected EOF
Peer 808 unexpectedly exited with status 1
`````
Two or three more days until release.

**Reply by netblue30** [reply]
OK! The 32 bit platform is fixed now. I'll play around a little bit with whitelist /dev

**Comment by ghost** [comment]
Anything holding up #5563 and #5571 from going in 0.9.72?

**Reply by netblue30** [reply]
merging now, add on top of them directly.

**Comment by netblue30** [comment]
Release Monday morning!

**Comment by netblue30** [comment]
More coming from here: https://github.com/netblue30/firejail/pull/5591, so probably no release on Monday.

**Comment by reinerh** [comment]
@netblue30 Can you please make a proper Github release?
The [releases page](https://github.com/netblue30/firejail/releases) still lists 0.9.72rc1 and 0.9.70.

**Reply by kmk3** [reply]
@reinerh [on Jan 16](https://github.com/netblue30/firejail/discussions/5192#discussioncomment-4699775):

> @netblue30 Can you please make a proper Github release?

From the page of the release itself, everything appears to be correct:

* <https://github.com/netblue30/firejail/releases/tag/0.9.72>

> The [releases page](https://github.com/netblue30/firejail/releases) still
> lists 0.9.72rc1 and 0.9.70.

It seems to be an issue with the Releases page (maybe it's cached?).

Note that 0.9.72 is marked as "Latest" on its own page and that there is
nothing else marked as "Latest" on the Releases page.

**Edit**: 0.9.72 also shows up as the current release on the home page:

* <https://github.com/netblue30/firejail>

**Reply by netblue30** [reply]
Probably a bug on GitHub site.

**Reply by kmk3** [reply]
Reported as a bug:

* <https://github.com/orgs/community/discussions/44718>

**Reply by kmk3** [reply]
After editing the release to add the new profiles, it appears on the releases
page:

* <https://github.com/netblue30/firejail/releases>

**Comment by kmk3** [comment]
I'm looking to close the 0.9.72 project so that only 0.9.74 appears when trying
to add an issue/PR to a project.

Links:

* [0.9.72](https://github.com/netblue30/firejail/projects/6)
* [0.9.74](https://github.com/netblue30/firejail/projects/7)

@rusty-snake I see that you added a few items to "To Do" on 0.9.72 (and I think
that the rest I copied from items you added on 0.9.70); could you help triage
them?  For example:

Items on "To Do" that were fixed on 0.9.72:

* Move it to "Done (on RELNOTES N/A)" on 0.9.72

Items on "To Do" that don't seem to be fixed but that are still relevant:

* Copy it to 0.9.74
* Archive it on 0.9.72
* If applicable, consider asking the author if it was fixed

Items on "To Do" that don't seem to be fixed and that are not relevant anymore:

* Archive it on 0.9.72

(If there's any item that doesn't apply, comment it in here and we'll figure it
out)

This should help ensure that things to be done from that project are not
forgotten.  And thus the project can be closed and ignored, which helps avoid
having to keep track of past projects (alongside the current project).

**Reply by rusty-snake** [reply]
> --tab (#4936) shell completions and vim syntax files

Was this originally from you or me?

> *Everything else in To Do

All have/are an issue.

**Reply by kmk3** [reply]
@rusty-snake [on Jan 21](https://github.com/netblue30/firejail/discussions/5192#discussioncomment-4744335):

> > --tab (#4936) shell completions and vim syntax files
> 
> Was this originally from you or me?

Now that you mention it, I'm not sure.

I had thought that it was added together with the keepassxc one (which is
archived on [0.9.70][1] and says that it was added by you), but it's not
archived in the older projects.

Anyway, looks like I added it; archived + copied to 0.9.74.

> > * Everything else in To Do
> 
> All have/are an issue.

To be clear, archiving an item hides it from the project card but keeps it
linked to the project.  The status of the issue/PR itself is not affected.

Note: The project archive can be accessed from:

* Menu -> ... -> View archive

By the way, we are the only ones that appear on the Activity feed of 0.9.72 and
the end of it says:

> This project was created before project activity started being fully
> recorded. There may be gaps in the beginning of this project’s activity.

I'm fairly sure I didn't add any of the remaining (profile-related) issues, so
if you didn't either, feel free to just archive them (or just say so and I'll
archive them).  If anyone else remembers adding any of them, feel free to copy
it to 0.9.74.

[1]: https://github.com/netblue30/firejail/projects/5

**Reply by rusty-snake** [reply]
With "All have/are an issue." I mean: do we need to track them in a project. IMHO we can just archive them.

**Reply by kmk3** [reply]
@rusty-snake [on Jan 21](https://github.com/netblue30/firejail/discussions/5192#discussioncomment-4744758):

> With "All have/are an issue." I mean: do we need to track them in a project.
> IMHO we can just archive them.

I personally don't mind just archiving these issues; what I meant was mainly
because I don't know why someone might have added any given issue to To Do, so
I'm not sure if anyone would care about just archiving it.

Maybe we could try to define what makes sense to put in the To Do of a release
project.  I'll open a separate discussion about it.
